plugins/synaprom/fu-synaprom-device.c

```diff
@@ -0,0 +1,504 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2019 Synaptics Inc
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-synaprom-firmware.h"
+#include "fu-synaprom-device.h"
+
+struct _FuSynapromDevice {
+	FuUsbDevice		 parent_instance;
+	guint8			 vmajor;
+	guint8			 vminor;
+};
+
+#define FU_SYNAPROM_PRODUCT_PROMETHEUS			65	/* Prometheus (b1422) */
+#define FU_SYNAPROM_PRODUCT_PROMETHEUSPBL		66
+#define FU_SYNAPROM_PRODUCT_PROMETHEUSMSBL		67
+
+/* vendor-specific USB control requets to write DFT word (Hayes) */
+#define FU_SYNAPROM_USB_CTRLREQUEST_VENDOR_WRITEDFT	21
+
+/* endpoint addresses for command and fingerprint data */
+#define FU_SYNAPROM_USB_REQUEST_EP			0x01
+#define FU_SYNAPROM_USB_REPLY_EP			0x81
+#define FU_SYNAPROM_USB_FINGERPRINT_EP			0x82
+#define FU_SYNAPROM_USB_INTERRUPT_EP			0x83
+
+#define FU_SYNAPROM_CMD_GET_VERSION			1
+#define FU_SYNAPROM_CMD_BOOTLDR_PATCH			125
+#define FU_SYNAPROM_CMD_IOTA_FIND			142
+
+typedef struct __attribute__((packed)) {
+	guint8		 command;
+} FuSynapromCmdGeneric;
```

Comment:
Isn't a packed struct this a bit much for just typecasting a single byte?

Reply:
This was copied from the Synaptics reference code. I'll nuke it.